### PR TITLE
update mgt-file-list with new attributes and options from file-upload.

### PR DESCRIPTION
**Pull request by aaclage**
update mgt-file-list with new attributes and options from new file-upload feature.

for revision @beth-panx

### CODE REVIEW

File: concepts/toolkit/components/file-list.md [hunk 1]
```diff
@@ -111,17 +148,17 @@ To learn more, see [styling components](../customize-components/style.md).
...
-| Pprovide `{user-id}` AND `{item-id}` | Files.Read, Files.Read.All, Sites.Read.All | `GET /users/{user-id}/drive/items/{item-id}/children` |
-| Provide `{drive-id}` AND `{item-path}` | Files.Read, Files.Read.All, Sites.Read.All | `GET /drives/{drive-id}/root:/{item-path}:/children` |
-| Provide `{group-id}` AND `{item-path}` | Files.Read, Files.Read.All, Sites.Read.All | `GET /groups/{group-id}/root:/{item-path}:/children` |
-| Provide `{site-id}` AND `{item-path}` | Files.Read, Files.Read.All, Sites.Read.All | `GET /sites/{site-id}/root:/{item-path}:/children` |
-| Provide `{user-id}` AND `{item-path}` | Files.Read, Files.Read.All, Sites.Read.All | `GET /users/{user-id}/root:/{item-path}:/children` |
-| Provide only `{item-path}` | Files.Read, Files.Read.All, Sites.Read.All | `GET /me/drive/root:/{item-path}:/children` |
+| Default (no identifiers or query provided) | Files.Read, Files.Read.All, Sites.Read.All, Files.ReadWrite, Files.ReadWrite.All, Sites.ReadWrite.All | `GET /me/drive/root/children` <br /> `PUT /me/drive/root:/{filename}:/content` <br /> `POST /me/drive/root:/{filename}:/createUploadSession` |
```
Comment (beth-panx) [inline concepts/toolkit/components/file-list.md:151]: @aaclage I think these need to be separated to their own row. They depend on the attribute `{enable-file-upload}`.
```suggestion
| Default (no identifiers or query provided) | Files.Read, Files.Read.All, Sites.Read.All | `GET /me/drive/root/children` |
| Provide `{enable-file-upload}` | Files.Read, Files.Read.All, Sites.Read.All, Files.ReadWrite, Files.ReadWrite.All, Sites.ReadWrite.All | `GET /me/drive/root/children` <br /> `PUT /me/drive/root:/{filename}:/content` <br /> `POST /me/drive/root:/{filename}:/createUploadSession` |
```

File: concepts/toolkit/components/file-list.md [hunk 2]
```diff
@@ -111,17 +148,17 @@ To learn more, see [styling components](../customize-components/style.md).
...
+| Provide `{drive-id}` AND `{item-id}` | Files.Read, Files.Read.All, Sites.Read.All, Files.ReadWrite, Files.ReadWrite.All, Sites.ReadWrite.All | `GET /drives/{drive-id}/items/{item-id}/children` <br /> `PUT /drives/{drive-id}/items/{item-id}:/{filename}:/content` <br /> `POST /drives/{drive-id}/items/{item-id}:/{filename}:/createUploadSession` |
```
Comment (beth-panx) [inline concepts/toolkit/components/file-list.md:152]: Similar here and below:

```suggestion
| Provide `{drive-id}` AND `{item-id}` | Files.Read, Files.Read.All, Sites.Read.All | `GET /drives/{drive-id}/items/{item-id}/children`|
| Provide `{drive-id}` AND `{item-id}` AND `{enable-file-upload}` | Files.Read, Files.Read.All, Sites.Read.All, Files.ReadWrite, Files.ReadWrite.All, Sites.ReadWrite.All | `GET /drives/{drive-id}/items/{item-id}/children` <br /> `PUT /drives/{drive-id}/items/{item-id}:/{filename}:/content` <br /> `POST /drives/{drive-id}/items/{item-id}:/{filename}:/createUploadSession` |
```

File: concepts/toolkit/components/file-list.md [hunk 3]
```diff
@@ -35,6 +35,10 @@ You can use several properties to customize the component.
 | page-size | pageSize | A number value to indicate the maximum number of files to render on each page. |
 | file-extensions | fileExtensions | An array of file extensions used to filter files to show. |
 | hide-more-files-button | hideMoreFilesButton | A boolean to indicate whether to show a button to render more files. |
+| enable-file-upload | enableFileUpload | Boolean to enable or disable file upload extension, default is false.  |
+| excluded-file-extensions | excludedFileExtensions | String array of file extensions to be excluded from file upload. |
+| max-file-size | maxFileSize | Number to restrict upload size (KB).  |
+| max-upload-file | maxUploadFile | Number, default value are 10 files. |
```
Comment (beth-panx) [inline concepts/toolkit/components/file-list.md:41]:
```suggestion
| excluded-file-extensions | excludedFileExtensions | String array of file extensions to be excluded from file upload. Must also provide `enable-file-upload` attribute. |
| max-file-size | maxFileSize | Number to restrict upload size (KB). Must also provide `enable-file-upload` attribute. |
| max-upload-file | maxUploadFile | Number, default value are 10 files. Must also provide `enable-file-upload` attribute. |
```

File: concepts/toolkit/components/file-list.md [hunk 4]
```diff
@@ -66,6 +70,30 @@ The following example changes the behavior of the component to fetch file list b
 <mgt-file-list insight-type="shared"></mgt-file-list>
 ```
 
+The following example enable the file upload feature.
```
Comment (beth-panx) [inline concepts/toolkit/components/file-list.md:73]:
```suggestion
The following example enables the file upload feature.
```

File: concepts/toolkit/components/file-list.md [hunk 5]
```diff
@@ -66,6 +70,30 @@ The following example changes the behavior of the component to fetch file list b
...
+
+```html
+<mgt-file-list enable-file-upload></mgt-file-list>
+```
+
+The following example limit upload to 5 files.
```
Comment (beth-panx) [inline concepts/toolkit/components/file-list.md:79]:
```suggestion
The following example limits maximum number of files that can be uploaded to 5.
```

File: concepts/toolkit/components/file-list.md [hunk 6]
```diff
@@ -66,6 +70,30 @@ The following example changes the behavior of the component to fetch file list b
...
+
+```html
+<mgt-file-list max-upload-file="5" enable-file-upload></mgt-file-list>
+```
+
+The following example limit upload file size to 10000 KB.
```
Comment (beth-panx) [inline concepts/toolkit/components/file-list.md:85]:
```suggestion
The following example limits maximum file size that can be uploaded to 10000 KB.
```

File: concepts/toolkit/components/file-list.md [hunk 7]
```diff
@@ -66,6 +70,30 @@ The following example changes the behavior of the component to fetch file list b
...
+
+```html
+<mgt-file-list max-file-size="10000" enable-file-upload></mgt-file-list>
+```
+
+The following example exclude upload of files with file extensions ".doc,.pdf".
```
Comment (beth-panx) [inline concepts/toolkit/components/file-list.md:91]:
```suggestion
The following example excludes upload of files with file extensions ".doc,.pdf".
```

**Review by beth-panx — CHANGES_REQUESTED** [review]
Need to change a few things. @elisenyang feel free to add your comments.

File: concepts/toolkit/components/file-list.md (hunk 2 above)
Comment (beth-panx) [reply concepts/toolkit/components/file-list.md:152]: Need to change the sections following this part too. :)

**Comment by aaclage** [comment]
> Need to change a few things. @elisenyang feel free to add your comments.

@beth-panx working on it.

**Comment by aaclage** [comment]
@beth-panx @elisenyang

API graph table updated.

**Review by beth-panx — APPROVED** [review]

File: concepts/toolkit/components/file-list.md [hunk 8]
```diff
@@ -35,6 +35,10 @@ You can use several properties to customize the component.
 | page-size | pageSize | A number value to indicate the maximum number of files to render on each page. |
 | file-extensions | fileExtensions | An array of file extensions used to filter files to show. |
 | hide-more-files-button | hideMoreFilesButton | A boolean to indicate whether to show a button to render more files. |
+| enable-file-upload | enableFileUpload | Boolean to enable or disable file upload extension, default is false.  |
```
Comment (elisenyang) [inline concepts/toolkit/components/file-list.md:38]:
```suggestion
| enable-file-upload | enableFileUpload | Boolean to enable or disable file upload functionality. The default value is `false`.  |
```

File: concepts/toolkit/components/file-list.md [hunk 9]
```diff
@@ -35,6 +35,10 @@ You can use several properties to customize the component.
...
+| excluded-file-extensions | excludedFileExtensions | String array of file extensions to be excluded from file upload. Must also provide `enable-file-upload` attribute. |
```
Comment (elisenyang) [inline concepts/toolkit/components/file-list.md:39]:
```suggestion
| excluded-file-extensions | excludedFileExtensions | String array of file extensions to be excluded from file upload. Must also set the `enable-file-upload` attribute to `true`. |
```

File: concepts/toolkit/components/file-list.md [hunk 10]
```diff
@@ -35,6 +35,10 @@ You can use several properties to customize the component.
...
+| max-file-size | maxFileSize | Number to restrict upload size (KB). Must also provide `enable-file-upload` attribute. |
```
Comment (elisenyang) [inline concepts/toolkit/components/file-list.md:40]:
```suggestion
| max-file-size | maxFileSize | A number representing the maximum file upload size (KB). Must also set the `enable-file-upload` attribute to `true`. |
```

File: concepts/toolkit/components/file-list.md [hunk 11]
```diff
@@ -35,6 +35,10 @@ You can use several properties to customize the component.
...
+| max-upload-file | maxUploadFile | Number, default value are 10 files. Must also provide `enable-file-upload` attribute. |
```
Comment (elisenyang) [inline concepts/toolkit/components/file-list.md:41]:
```suggestion
| max-upload-file | maxUploadFile | A number representing the maximum number of files allowed to be uploaded. The default value is `10` files. Must also set the `enable-file-upload` attribute to `true`. |
```

File: concepts/toolkit/components/file-list.md [hunk 12]
```diff
@@ -66,6 +70,30 @@ The following example changes the behavior of the component to fetch file list b
 <mgt-file-list insight-type="shared"></mgt-file-list>
 ```
 
+The following example enables the file upload feature.
+
+```html
+<mgt-file-list enable-file-upload></mgt-file-list>
+```
+
+The following example limits maximum number of files that can be uploaded to 5.
```
Comment (elisenyang) [inline concepts/toolkit/components/file-list.md:79]:
```suggestion
The following example limits the maximum number of files that can be uploaded to 5.
```

File: concepts/toolkit/components/file-list.md [hunk 13]
```diff
@@ -66,6 +70,30 @@ The following example changes the behavior of the component to fetch file list b
...
+
+```html
+<mgt-file-list max-upload-file="5" enable-file-upload></mgt-file-list>
+```
+
+The following example limits maximum file size that can be uploaded to 10000 KB.
```
Comment (elisenyang) [inline concepts/toolkit/components/file-list.md:85]:
```suggestion
The following example limits the maximum file size that can be uploaded to 10000 KB.
```

**Review by elisenyang — APPROVED** [review]

**Comment by aaclage** [comment]
@elisenyang Thanks

**Pull request merged**